src/latch_cli/services/k8s/develop.py

```diff
@@ -0,0 +1,283 @@
+from latch_sdk_gql.execute import execute
+
+from ...constants import docker_image_name_illegal_pat, latch_constants
+from ...utils import (
+    TemporarySSHCredentials,
+    get_auth_header,
+    identifier_suffix_from_str,
+)
+from .ws_utils import forward_stdio
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+max_polls = 1800
+
+
+# not going to allow most of these for now
+class InstanceSize(str, Enum):
+    small_task = "small_task"
+    medium_task = "medium_task"
+    # large_task = "large_task"
+    small_gpu_task = "small_gpu_task"
+    # large_gpu_task = "large_gpu_task"
+    # v100_x1_task = "v100_x1_task"
+    # v100_x4_task = "v100_x4_task"
+    # v100_x8_task = "v100_x8_task"
+    # g6e_xlarge_task = "g6e_xlarge_task"
+    # g6e_2xlarge_task = "g6e_2xlarge_task"
+    # g6e_4xlarge_task = "g6e_4xlarge_task"
+    # g6e_8xlarge_task = "g6e_8xlarge_task"
+    # g6e_12xlarge_task = "g6e_12xlarge_task"
+    # g6e_16xlarge_task = "g6e_16xlarge_task"
+    # g6e_24xlarge_task = "g6e_24xlarge_task"
+    # g6e_48xlarge_task = "g6e_48xlarge_task"
+
+
+def workflow_name(pkg_root: Path) -> str:
+    name_path = pkg_root / latch_constants.pkg_workflow_name
+    if not name_path.exists():
+        click.secho(
+            "Unable to parse workflow name - please make sure you have registered your workflow first.",
+            fg="red",
+        )
+        raise click.exceptions.Exit(1)
+
+    return name_path.read_text().strip()
+
+
+def get_image(pkg_root: Path) -> str:
+    ws_id = current_workspace()
+    wf_name = workflow_name(pkg_root)
+
+    prefix = ws_id
+    if int(ws_id) < 10:
+        prefix = f"x{ws_id}"
+
+    suffix = wf_name
+    suffix = identifier_suffix_from_str(wf_name).lower()
+    suffix = docker_image_name_illegal_pat.sub("_", wf_name)
+
+    return f"{prefix}_{suffix}"
+
+
+def get_latest_registered_version(pkg_root: Path) -> str:
+    ws_id = current_workspace()
+
+    wf_name = workflow_name(pkg_root)
+
+    res = execute(
+        gql.gql("""
+        query LatestVersion($wsId: BigInt!, $name: String!) {
+            workflowInfosLatestVersionInAccount(
+                argOwnerId: $wsId
+                filter: {name: {equalTo: $name}}
+            ) {
+                nodes {
+                    version
+                }
+            }
+        }
+        """),
+        {"wsId": ws_id, "name": wf_name},
+    )["workflowInfosLatestVersionInAccount"]
+
+    if len(res["nodes"]) != 1:
+        click.secho(
+            "Unable to find a registered workflow version - please make sure you have registered your workflow first.",
+            fg="red",
+        )
+        raise click.exceptions.Exit(1)
+
+    return res["nodes"][0]["version"]
+
+
+async def rsync(pkg_root: Path, ip: str):
+    ssh_command = "ssh -o CheckHostIP=no -o StrictHostKeyChecking=no -o ServerAliveInterval=30 -o ServerAliveCountMax=5 -J root@a1fd7b5d7b5824b46b8671207e1124b6-610912349.us-west-2.elb.amazonaws.com"
```

Comment:
wanna drop the elb into a config or a constant? Idrc but would prob seem better as "ssh_forwarder_permanent_address" or smtg